PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/action/SelectSparkApplicationTypeAction.kt

```diff
@@ -42,6 +47,16 @@ abstract class SelectSparkApplicationTypeAction
             if (!DefaultLoader.getIdeHelper().isApplicationPropertySet(CommonConst.SPARK_APPLICATION_TYPE)) return SparkApplicationType.None
             return SparkApplicationType.valueOf(DefaultLoader.getIdeHelper().getApplicationProperty(CommonConst.SPARK_APPLICATION_TYPE))
         }
+
+        @JvmStatic
+        fun getRunConfigurationType() : ConfigurationType? {
+            return when(getSelectedSparkApplicationType()) {
+                SparkApplicationType.None -> null
+                SparkApplicationType.HDInsight -> LivySparkBatchJobRunConfigurationType.getInstance()
+                SparkApplicationType.CosmosSpark -> CosmosSparkConfigurationType
+                SparkApplicationType.CosmosServerlessSpark -> CosmosServerlessSparkConfigurationType
+            }

```

Comment:
Do you need an `else -> null`?

Reply:
All possible cases are covered in the branches. else branch is not necessary.